### ISSUE TITLE
json.path.read max() returns a string of a double

### ISSUE THREAD

**Issue by aliasmask** [RPTools/maptool#1326]
**Describe the bug**
When using max() function in json.path.read it returns a string of a double and without knowing about it, blows up other functions requiring a number.

**To Reproduce**
```
[H: a = json.set("{}","mods", '[
            {
                "name": "mod1",
                "type": "type1",
                "value": 1
            },
            {
                "name": "mod2",
                "type": "type1",
                "value": 2
            },
            {
                "name": "mod3",
                "type": "type1",
                "value": 3
            },
            {
                "name": "mod4",
                "type": "type2",
                "value": 4
            }]'
)]

[H: values = json.path.read(a,"mods..[?(@.type == 'type1')]value")]
[R: json.path.read(values,"max()")]
```

I get this result: 3.0
So, when I do this 
`[R: floor(json.path.read(values,"max()"))]`

I get this error 
**_Illegal argument type java.lang.String, expecting java.math.BigDecimal_**

Which tells me it's returning a string of a double value and requires me to do this 
`[R: floor(number(json.path.read(values,"max()")))]`

**Expected behavior**
I would expect it to actually return an integer, but the specs say double, so I guess I'm okay with that, but it definitely shouldn't be a string.

**MapTool Info**
- Version: 1.5.13

**Additional context**
I wish there was more documentation on the use of functions for json.path.read.

**Comment by aliasmask** [comment]
You may want to check the other functions like min() and avg() as well for the same issue.

**Comment by Merudo** [comment]
I'll take care of this one

**Comment by Merudo** [comment]
@Phergus beat me to the punch :P

The issue can be fixed by editing `asScriptType` in `JsonMTSTypeConversion`.

**Comment by cwisniew** [comment]
} else if (val instanceof Double) {
      return BigDecimal.valueOf((Double) val);
    } else if (val instanceof Integer) {
      return BigInteger.valueOf((Integer) val);
    }

**Comment by Phergus** [comment]
Yes.  I already did that.  Just hadn't got around to pushing it up as I was making dinner.

**Comment by Phergus** [comment]
Released in 1.5.14.  Closing.
